### ISSUE TITLE
 java.lang.NoSuchMethodError while invoking hc.avroFile(inputDir)

### ISSUE THREAD

**Issue by deepujain** [databricks/spark-avro#37]
package com.ep.poc.spark.reporting.process.service

import com.ep.poc.spark.reporting.process.util.DateUtil._

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

import collection.mutable.HashMap

import com.databricks.spark.avro._

class HadoopSuccessEvents2Service extends ReportingService {

  override def execute(arguments: HashMap[String, String], sc: SparkContext) {
    val detail = "reporting.detail." + arguments.get("subcommand").get
    val startDate = formatStringAsDate(arguments.get("startDate").get)
    val endDate = formatStringAsDate(arguments.get("endDate").get)
    val inputDir = arguments.get("input").get
    val outputDir = arguments.get("output").get

    val hc = new org.apache.spark.sql.hive.HiveContext(sc)
    val successDetail_S1 = hc.avroFile(inputDir)
    successDetail_S1.registerTempTable("success_events.sojsuccessevents1")
    val countS1 = hc.sql("select count(*) from success_events.sojsuccessevents1")
    println(countS1.collect)
  }
}

I am sorry to raise it as a issue. Can you share any mailing list or google group  ?

Logs
====
Exception in thread "Driver" scala.MatchError: java.lang.NoSuchMethodError: com.databricks.spark.avro.package$AvroContext.avroFile(Ljava/lang/String;)Lorg/apache/spark/sql/DataFrame; (of class java.lang.NoSuchMethodError)

**Comment by vlyubin** [comment]
Hm, that's weird. Just to get some context - this happens with the latest version of master (or some older version and Spark 1.2)?

I just checked out spark-avro, built it (`sbt/sbt package`) and launched a 1.3 spark-shell with the JAR that I got:
`spark-shell --jars target/scala-2.10/spark-avro_2.10-1.0.0.jar`
Then I ran the following commands and it seemed to work:
```import com.databricks.spark.avro._
val hc = new org.apache.spark.sql.hive.HiveContext(sc)
hc.avroFile("src/test/resources/episodes.avro").collect()
```

Can you let me know if this works for you?

**Comment by deepujain** [comment]
Spark Version: 1.2.1
Spark-Avro - 
			<groupId>com.databricks</groupId>
			<artifactId>spark-avro_2.10</artifactId>
			<version>0.1</version>

**Comment by vlyubin** [comment]
Ah, so you were using 1.2

Sorry about that, we should have mentioned it on the README before - 0.1 is a very OLD build of spark-avro, and it doesn't include all the additions that were made to master and mentioned in the README.

Anyway, as spark 1.3 was released, we made to releases of this package:
`0.2.0` - spark-avro for Spark 1.2
`1.0.0` - spark-avro for Spark 1.3

Can you change 0.1 to 0.2.0 and see if it works?

P.S. Readme is updated as well.

**Comment by vlyubin** [comment]
Actually, it looks like they won't actually be on maven for a couple of days, so you have to wait a little bit ...

**Comment by vlyubin** [comment]
Ok, it's on maven now. Would you mind telling if the issue is resolved?

Thanks :)

**Comment by deepujain** [comment]
I was off for a while. I am back now.
When i ran with the new Jar with Spark 1.3 i ran into this

 ... 27 more

Caused by: java.lang.ClassNotFoundException:
org.datanucleus.api.jdo.JDOPersistenceManagerFactory

at java.net.URLClassLoader$1.run(URLClassLoader.java:366)

at java.net.URLClassLoader$1.run(URLClassLoader.java:355)

at java.security.AccessController.doPrivileged(Native Method)

at java.net.URLClassLoader.findClass(URLClassLoader.java:354)

at java.lang.ClassLoader.loadClass(ClassLoader.java:425)

at java.lang.ClassLoader.loadClass(ClassLoader.java:358)

at java.lang.Class.forName0(Native Method)

at java.lang.Class.forName(Class.java:270)

at javax.jdo.JDOHelper$18.run(JDOHelper.java:2018)

at javax.jdo.JDOHelper$18.run(JDOHelper.java:2016)

at java.security.AccessController.doPrivileged(Native Method)

at javax.jdo.JDOHelper.forName(JDOHelper.java:2015)

at
javax.jdo.JDOHelper.invokeGetPersistenceManagerFactoryOnImplementation(JDOHelper.java:1162)

... 46 more


From my command prompt am able to run hive

-sh-4.1$ hive

15/03/24 02:31:17 WARN conf.HiveConf: DEPRECATED: Configuration property
hive.metastore.local no longer has any effect. Make sure to provide a valid
value for hive.metastore.uris if you are connecting to a remote metastore.

15/03/24 02:31:17 WARN conf.HiveConf: DEPRECATED: hive.metastore.ds.retry.*
no longer has any effect.  Use hive.hmshandler.retry.* instead

WARNING: org.apache.hadoop.metrics.jvm.EventCounter is deprecated. Please
use org.apache.hadoop.log.metrics.EventCounter in all the log4j.properties
files.


Logging initialized using configuration in
file:/etc/hdp/apollo-hdp213/hive/conf/hive-log4j.properties

hive>

On Mon, Mar 23, 2015 at 5:24 AM, Volodymyr Lyubinets <
notifications@github.com> wrote:

> Ok, it's on maven now. Would you mind telling if the issue is resolved?
>
> Thanks :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/databricks/spark-avro/issues/37#issuecomment-84734278>
> .
>



-- 
Deepak

**Comment by deepujain** [comment]
http://mail-archives.apache.org/mod_mbox/spark-user/201408.mbox/%3CCANAZ_dGFMEioMR-rwzQzyDSk1jwErx_yzZwvik3bcv0gF=KVHQ@mail.gmail.com%3E

Any idea how to fix this ? Which JARS do i need to include with --jars in
addition to spark-avro for it to get running.

On Tue, Mar 24, 2015 at 3:02 PM, ÐΞ€ρ@Ҝ (๏̯͡๏) <deepujain@gmail.com> wrote:

> I was off for a while. I am back now.
> When i ran with the new Jar with Spark 1.3 i ran into this
>
>  ... 27 more
>
> Caused by: java.lang.ClassNotFoundException:
> org.datanucleus.api.jdo.JDOPersistenceManagerFactory
>
> at java.net.URLClassLoader$1.run(URLClassLoader.java:366)
>
> at java.net.URLClassLoader$1.run(URLClassLoader.java:355)
>
> at java.security.AccessController.doPrivileged(Native Method)
>
> at java.net.URLClassLoader.findClass(URLClassLoader.java:354)
>
> at java.lang.ClassLoader.loadClass(ClassLoader.java:425)
>
> at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
>
> at java.lang.Class.forName0(Native Method)
>
> at java.lang.Class.forName(Class.java:270)
>
> at javax.jdo.JDOHelper$18.run(JDOHelper.java:2018)
>
> at javax.jdo.JDOHelper$18.run(JDOHelper.java:2016)
>
> at java.security.AccessController.doPrivileged(Native Method)
>
> at javax.jdo.JDOHelper.forName(JDOHelper.java:2015)
>
> at
> javax.jdo.JDOHelper.invokeGetPersistenceManagerFactoryOnImplementation(JDOHelper.java:1162)
>
> ... 46 more
>
>
> From my command prompt am able to run hive
>
> -sh-4.1$ hive
>
> 15/03/24 02:31:17 WARN conf.HiveConf: DEPRECATED: Configuration property
> hive.metastore.local no longer has any effect. Make sure to provide a valid
> value for hive.metastore.uris if you are connecting to a remote metastore.
>
> 15/03/24 02:31:17 WARN conf.HiveConf: DEPRECATED:
> hive.metastore.ds.retry.* no longer has any effect.  Use
> hive.hmshandler.retry.* instead
>
> WARNING: org.apache.hadoop.metrics.jvm.EventCounter is deprecated. Please
> use org.apache.hadoop.log.metrics.EventCounter in all the log4j.properties
> files.
>
>
> Logging initialized using configuration in
> file:/etc/hdp/apollo-hdp213/hive/conf/hive-log4j.properties
>
> hive>
>
> On Mon, Mar 23, 2015 at 5:24 AM, Volodymyr Lyubinets <
> notifications@github.com> wrote:
>
>> Ok, it's on maven now. Would you mind telling if the issue is resolved?
>>
>> Thanks :)
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/databricks/spark-avro/issues/37#issuecomment-84734278>
>> .
>>
>
>
>
> --
> Deepak
>
>


-- 
Deepak

**Comment by vlyubin** [comment]
So from the mailing list post is seems like you were able to solve this issue. Is that correct?

**Comment by vlyubin** [comment]
I assume the problem is over? Sorry again for the confusion with `0.1`.

**Comment by deepujain** [comment]
Yes

On Thu, Apr 9, 2015 at 6:22 AM, Volodymyr Lyubinets <
notifications@github.com> wrote:

> Closed #37 <https://github.com/databricks/spark-avro/issues/37>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/databricks/spark-avro/issues/37#event-276768810>.
>



-- 
Deepak
